src/wallet/wallet2.cpp

```diff
@@ -3063,6 +3073,26 @@ bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_
     m_account.set_device(hwdev);
     LOG_PRINT_L0("Device inited...");
   }
+
+  if (r)
+  {
+    if (encrypted_secret_keys)
+    {
+      m_account.decrypt_keys(key);
+    }
+    else
+    {
+      // rewrite with encrypted keys, ignore errors
+      encrypt_keys(key);
+      bool saved_ret = store_keys(keys_file_name, password, m_watch_only);
+      if (!saved_ret)
+      {
+        // just moan a bit, but not fatal
+        MERROR("Error saving keys file with encrypted keys, not fatal");
+      }
+      decrypt_keys(key);
```

Comment:
m_account.decrypt_keys(key);

Reply:
After a bit of testing and looking closely, it seems like these `encrypt_keys`/`decrypt_keys` steps here are unnecessary, because the function `store_keys` internally calls `encrypt_keys` against a copy of `m_account`.

Reply:
It's a copy, but made at the start of it, so it needs to start from the expected state, which is encrypted secret spend key.